### PR TITLE
#2204

### PR DESCRIPTION
**Pull request by wsbrenk**
1. changes proposed in this pull request:

   - fixes issue #2204
   - if not fixing an existing issue, comments explaining the purpose of the PR should be provided)

2. `src/main/resources/release_notes.md` ...

 - [x] has been updated
 - [ ] does not require update

3. [Optional] suggested person to review this PR @___

### CODE REVIEW

File: src/test/java/core/util/HOCurrencyTests.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package core.util;
```
Comment (tychobrailleur) [inline src/test/java/core/util/HOCurrencyTests.java:1]: I am intrigued as to why you didn't write the tests in Kotlin? :-)

File: src/main/java/core/model/player/Player.java [hunk 2]
```diff
@@ -2462,4 +2459,42 @@ public long calculateTSI() {
...
+        if ( experience >= trainerSkill && trainerSkill >= 4 && trainerSkill < 9){
+            if ( trainerSkill >= 5){
+                var a = switch (trainerSkill) {
+                    case 5 -> 0.037;
+                    case 6 -> 0.296;
+                    case 7 -> 1;
+                    case 8 -> 7.11;
+                    default -> throw new IllegalStateException("Unexpected value: " + trainerSkill);
+                };
+                return new HOCurrency((int)(47500000 * a / (experience-1)));
```
Comment (tychobrailleur) [inline src/main/java/core/model/player/Player.java:2491]: In my old age, I find it hard to parse numbers above thousands.  Would you be able to write `47_500_000` for clarity?  Same for `100_000` below.

File: src/main/java/core/gui/model/UserColumnFactory.java [hunk 3]
```diff
@@ -948,7 +948,48 @@ public IHOTableEntry getTableEntry(Player player, Player playerCompare) {
...
         return playerAdditionalArray;
     }
+
+    private static IHOTableEntry getTrainerTransferEntry(Player player, int i) {
+        var costs = player.calculateCoachConversionCosts(i);
+        var string = "";
+        var swedishKrona = 0;
+        if (costs != null){
+            string = costs.toLocaleString();
+            swedishKrona = costs.getSwedishKrona();
```
Comment (tychobrailleur) [inline src/main/java/core/gui/model/UserColumnFactory.java:991]: I must admit I had no idea that the Swedish Krona was used as the basis for that calculation!

File: src/test/java/core/util/HOCurrencyTests.java (hunk 1 above)
Comment (wsbrenk) [reply src/test/java/core/util/HOCurrencyTests.java:1]: Step by step - first i want to see, how to use it in java code ;-)

I will change this.

File: src/main/java/core/model/player/Player.java (hunk 2 above)
Comment (wsbrenk) [reply src/main/java/core/model/player/Player.java:2491]: Thanks - again i learned new java stuff.

File: src/main/java/core/gui/model/UserColumnFactory.java (hunk 3 above)
Comment (wsbrenk) [reply src/main/java/core/gui/model/UserColumnFactory.java:991]: The formula was published using Euro, but the hattricks internally are using swedish krona in all xml files with a total unrealistic exchange course of 10 sk to 1 €;-)

Additionally all prices will be rounded to 100 sk respectively 10 €. 

These things i wanted to handle in one single utility class and no longer in several places with different approaches.

Comment (tychobrailleur) [reply src/main/java/core/gui/model/UserColumnFactory.java:991]: Thanks, again I have learnt something!! 😊

**Review by tychobrailleur — APPROVED** [review]
